lib/src/utils.dart

```diff
@@ -114,3 +115,46 @@ Logic condFlop(
 
   return (first, second);
 }
+
+/// This class is used to check the SystemVerilog output of a module for
+/// poor naming conventions, such as names with `__` in them. It can also
+/// generate a SystemVerilog file for the module if requested.
+class CheckModuleNaming {
```

Comment:
Why is this a class?  Is this something we want to expose as part of the package or just an internal utility?  This file is exported.

Reply:
It should be for testing only.  I'll label it.

Basically what this PR proposes is that for each module, we should write an instantiation test (and I discovered trace errors this way).  Just instantiate your module in its various configurations and check the names.  That would be a very quick test and would first make sure your instantiation works (no trace errors), and that the Verilog names can be checked and cleaned up.  you can see the module output and tune it to be more readable.

Reply:
It is a class because it holds a global static variable to increment a number to append to a filename.  So if we create multiple variations of a module and write them out, the filenames do not collide.  

I was about to add the 'test instantiation' tests that do variations of parameterization on a component to make sure new names don't pop up under different variations.  In that case it is useful to turn on file writing as well.